daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -959,32 +990,45 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
 
   protected def checkDiagnosticMessages(
     diagnostics: Seq[Diagnostic],
-    errors: ExpectedErrors,
-    optWarnings: Option[ExpectedWarnings],
+    errors: Seq[ExpectedErrors],
```

Comment:
Yes, making that change will allow us to call checkDiagnosticMessages everywhere we currently call verifyDiagnosticsFound. We would just pass in None for what we don't wish to check.

Ex to check only warnings or only validationErrors, we do
```
        checkDiagnosticMessages(
          allDiags,
          None,
          optWarnings,
          None,
          implString
        )
    //or 
        checkDiagnosticMessages(
          actual.getDiagnostics,
          None,
          None,
          optValidationErrors,
          implString
        )
    
```

Reply:
Seems reasonable. But are there cases where we don't what to check all expected diagnostics? Seems like during each test we do a parse/unparse/whatever and then check that all actual diagnostics match the expected diagnostics.

Reply:
I think so, in verifyParseResult for example, it looks like we check validation errors separate from warnings

```
(shouldValidate, expectsValidationError) match {
      case (_, true) => {
        // Note that even if shouldValidate is false, we still need to check
        // for validation diagnostics because failed assertions with
        // failureType="recoverableError" are treated as validation errors
        checkDiagnosticMessages(
          actual.getDiagnostics,
          None,
          None,
          optExpectedValidationErrors,
          implString
        )
        Assert.invariant(actual.isValidationError)
      }
      case (true, false) => {
        VerifyTestCase.verifyNoValidationErrorsFound(
          actual,
          implString
        ) // Verify no validation errors from parser
        Assert.invariant(!actual.isValidationError)
      }
      case (false, false) => // Nothing to do here.
    }

    val allDiags = compileWarnings ++ actual.getDiagnostics
    checkDiagnosticMessages(
      allDiags,
      None,
      optExpectedWarnings,
      None,
      implString
    )
```

Reply:
I imagine some of that can be refactored and simplified now that you've cleaned up much of this and made things more generic and combined functionality. Seems all three of those checks can be done with one call to checkDiagnosticMessages.

I guess we do still need `None` sometimes, since some functions (e.g. runParseExpectSuccess) don't have access to certain diagnostics because they expect success and really do need to pass in None. I think maybe the separate expectSuccess/expectFailure funtions are complicating things a bit, but that clean up can be done for another day.